### PR TITLE
Prevent rewrite flushes for runtime checks

### LINKED ISSUE #330: Plugin Check plugin will remove code in .htaccess
When I run the Plugin Check plugin on any plugin, the entire part in the `.htaccess` file between `#BEGIN WordPress` and `#END WordPress` is removed.
I mean the code from `<IfModule>` to `</IfModule>`.
I need to open the Settings > Permalinks page, which regenerates the missing rows.

Edit: This happen when the permalink structure is set to "Post name"

### PR DESCRIPTION
**Pull request by swissspidy**
Fixes #330

To test:

1. Activate a plugin
2. Set up pretty permalinks
3. Run performance checks for the activated plugin
4. See that `.htaccess` file won't be empty.

Cannot be tested with unit tests as there is no Apache with mod_rewrite in a testing environment, so it's not possible to verify `.htaccess` modifications in any way.

### CODE REVIEW

File: includes/Checker/Runtime_Environment_Setup.php [hunk 1]
```diff
@@ -27,11 +27,27 @@ public function set_up() {
...
 		if ( $wpdb->posts !== $wpdb->get_var( $wpdb->prepare( 'SHOW TABLES LIKE %s', $wpdb->posts ) ) ) {
+			/*
+			 * Set the same permalink structure *before* install finishes,
+			 * so that wp_install_maybe_enable_pretty_permalinks() does not flush rewrite rules.
+			 *
+			 * See https://github.com/WordPress/plugin-check/issues/330
+			 */
+			add_action(
+				'populate_options',
+				static function () use( $permalink_structure ) {
```
Comment (westonruter) [inline includes/Checker/Runtime_Environment_Setup.php]: Micro nit:

```suggestion
				static function () use ( $permalink_structure ) {
```

There is always a space after `use` in the core codebase.

Comment (swissspidy) [reply includes/Checker/Runtime_Environment_Setup.php]: Hmm this kinda stuff should be handled by WPCS :/

Comment (felixarntz) [reply includes/Checker/Runtime_Environment_Setup.php]: Also required by WPCS 3+ I think.

**Review by felixarntz — APPROVED** [review]
Thanks @swissspidy, great catch!

File: includes/Checker/Runtime_Environment_Setup.php (hunk 1 above)
Comment (swissspidy) [reply includes/Checker/Runtime_Environment_Setup.php]: Aren‘t we on WPCS 3 already?

**Review by westonruter — APPROVED** [review]

File: includes/Checker/Runtime_Environment_Setup.php (hunk 1 above)
Comment (westonruter) [reply includes/Checker/Runtime_Environment_Setup.php]: Yes:

https://github.com/WordPress/plugin-check/blob/35eee0bbbceaeb5bbd7281e185d7b65d3f03c7ef/composer.json#L10

Comment (westonruter) [reply includes/Checker/Runtime_Environment_Setup.php]: I've reported this upstream: https://github.com/WordPress/WordPress-Coding-Standards/issues/764#issuecomment-1851018615

Comment (westonruter) [reply includes/Checker/Runtime_Environment_Setup.php]: Already PR opened to fix! https://github.com/WordPress/WordPress-Coding-Standards/pull/2415

**Comment by mukeshpanchal27** [comment]
Thanks @swissspidy. Just miss to approve the PR last night. Great work.

**Comment by codestylist** [comment]
Thanks to everyone who fixed this small but important issue ;-). Good job!

**Pull request merged**